src/util/file-io/file-io.ts

```diff
@@ -125,6 +131,39 @@ export class FileIO implements IFileIO {
     return releases;
   }
 
+
+/**
+ * Discover, Compile and load external plugins
+ * from the plugins/ directory.
+ * 
+ * Returns 0..n plugins.
+ */
+   public GetAvailablePlugins(_pluginsToActivate: string[]): AbstractPlugin[]
+    {
+  // Directory in which to find plugins.
+  const DIRECTORY: string = "plugins/";
+
+  // Plugin directories
+  let directories: string[] = (this.fs.readdirSync(DIRECTORY).filter((f: any) => this.fs.statSync(DIRECTORY + "/" + f).isDirectory()));
+
+  // Get active plugins
+  let activePlugins: string[] = directories.filter(pluginDir => this.fs.existsSync(`${DIRECTORY}/${pluginDir}/plugin.ts`) && _pluginsToActivate.indexOf(pluginDir) > -1);
+  // Compile
+  // Get all directories with plugin.ts
+  // Rewrite Directory -> Directory/plugin.ts & Compile
+  (ts.createProgram(activePlugins
+    .map(pluginDir => `${DIRECTORY}${pluginDir}/plugin.ts`), {})).emit();
+
+  // Load & Return plugins.
+  return activePlugins
```

Comment:
Consider breaking this daunting one-liner up by making it multi-line and abstracting parts of it away behind functions.